### PR TITLE
refactor: downloaders rework

### PR DESCRIPTION
**Pull request by davidemarcoli**
# Pull Request Check List

Resolves: #issue-number-here

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

## Summary by CodeRabbit

- **New Features**
	- Enhanced job monitoring for service execution times, with logging for jobs exceeding three minutes.
	- New method for extracting info hash from magnet URIs, improving session management.
	- Added classes and methods for improved handling of video downloads and torrent management.
	- Introduced a timeout mechanism for request handling to enhance error management.

- **Bug Fixes**
	- Improved error handling for session management and torrent availability checks.
	- Enhanced clarity of validation error messages in settings management.

- **Documentation**
	- Updated logging and error messages for better traceability and user feedback.

- **Refactor**
	- Simplified type assignment in media classes, removing unnecessary enum abstractions.
	- Streamlined downloader interfaces and improved request handling logic.

- **Chores**
	- Removed unused imports and cleaned up codebase for better maintainability.

### CODE REVIEW

File: src/program/services/downloaders/torbox.py [hunk 1]
```diff
@@ -332,3 +332,264 @@
...
+    DOWNLOADING = "downloading"
+    DOWNLOADED = "downloaded"
+    ERROR = "error"
+    SEEDING = "seeding"
+    DEAD = "dead"
+    UPLOADING = "uploading"
+    COMPRESSING = "compressing"
+
+class TBTorrent(BaseModel):
+    """Real-Debrid torrent model"""
```
Comment (dreulavelle) [inline src/program/services/downloaders/torbox.py:371]: :D

Comment (davidemarcoli) [reply src/program/services/downloaders/torbox.py:371]: Uuuups 😆

**Comment by dreulavelle** [comment]
still working on it but its coming along pretty well.. ran into a snag though..
```log
24-11-23 23:49:47 | ❌ ERROR     | torbox.get_torrent_info - Failed to get torrent info for 182010: Request failed: 404 Client Error: Not Found for url: https://api.torbox.app/v1/api/torrents/torrentinfo/182010
24-11-23 23:49:47 | 🐞 DEBUG     | __init__.run - Invalid stream: 8b0ca81c975d347fb921b757c2dc593f01a5f794 - reason: Request failed: 404 Client Error: Not Found for url: https://api.torbox.app/v1/api/torrents/torrentinfo/182010
```

looks like the get torrent info endpoint isnt right for Torbox.. @davidemarcoli 

I'll work on it more when I wake up.. been at it for hours already lol

**Review by Gaisberg — APPROVED** [review]
LGTM!

**Pull request merged**